### PR TITLE
Added large tests, bug fixes

### LINKED ISSUE #17: Add large integration tests
We should check in some large header files (in third_party) + yaml config + expected outcome and run them as tests. (For example libclang.)

These large integration tests that will flag when something has changed.

Also, this will help us notice if there are any discrepancies between generated bindings on different OSes.

### LINKED ISSUE #29: Unsupported operation error
Probably my mistake here, but I'm getting an error I don't know how to resolve:

```
[C:\src\ffigen\example\c_json] dart pub run ffigen
Running in Directory: 'C:\src\ffigen\example\c_json'
Unhandled exception:
Unsupported operation: Illegal character in path: ..\tool\wrapped_libclang
#0      _Uri._checkWindowsPathReservedCharacters (dart:core/uri.dart:1734:11)
#1      _Uri._toWindowsFilePath (dart:core/uri.dart:2745:7)
#2      _Uri.toFilePath (dart:core/uri.dart:2720:38)
#3      Config._getSpecs.<anonymous closure> (package:ffigen/src/config_provider/config.dart:157:14)
#4      Config._extract (package:ffigen/src/config_provider/config.dart:134:49)
#5      new Config.fromYaml (package:ffigen/src/config_provider/config.dart:97:17)
#6      getConfigFromPubspec (file:///C:/src/ffigen/bin/ffigen.dart:72:17)
#7      getConfig (file:///C:/src/ffigen/bin/ffigen.dart:45:12)
#8      main (file:///C:/src/ffigen/bin/ffigen.dart:24:18)
#9      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#10     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

## dart info
```
[C:\src\ffigen\example\c_json] dart --version
Dart SDK version: 2.9.0-20.0.dev.flutter-06cb010247 (be) (Tue Jul 7 08:14:51 2020 +0000) on "windows_x64"
```

### PR DESCRIPTION
**Pull request by mannprerak2**
closes #29, closes #17

- Added large tests for Libclang, cJSON and SQLite #17.
- Bug fix: Nested parsing, added tests for nested parsing. Sub parsers now use a stack to hold information about current declaration.
- Bug fix: Unsupported operation error when finding dynamic library path on windows #29.
- Added property originalName to Binding.
- Config provider now throws ConfigError in-case a required config is missing instead. Fixes tests not running previously due to exit call in config provider.
- Updated libclang wrapper to support running from multiple isolates (fixes tests).

### CODE REVIEW

**Comment by mannprerak2** [comment]
Re-created #30 because of a PR tracking issue.

File: bin/ffigen.dart [hunk 1]
```diff
@@ -11,7 +11,7 @@ import 'package:ffigen/src/header_parser.dart' as parser;
 import 'package:logging/logging.dart';
 import 'package:yaml/yaml.dart' as yaml;
 
-var _logger = Logger('ffigen.dart');
+var _logger = Logger('ffigen.ffigen');
```
Comment (dcharkes) [inline bin/ffigen.dart:14]: Shouldn't this be `Logger('ffigen')`, such that it becomes the root of the other loggers?

File: lib/src/config_provider/config.dart [hunk 2]
```diff
@@ -153,7 +150,8 @@ class Config {
             'Path to folder containing libclang dynamic library, used to parse C headers',
         requirement: Requirement.no,
         defaultValue: () => getDylibPath(Platform.script
-            .resolve(path.join('..', 'tool', 'wrapped_libclang'))
+            .resolve(path.posix.join('..', 'tool',
+                'wrapped_libclang')) // Path needs to be in posix style here.
```
Comment (dcharkes) [inline lib/src/config_provider/config.dart:154]: Either explain why it needs to be posix style (Windows issues), or remove the comment.

File: bin/ffigen.dart (hunk 1 above)
Comment (mannprerak2) [reply bin/ffigen.dart:14]: This should be the root logger as this is the ffigen.dart script which is just another part of ffigen.
The root logger is created automatically in the Logger constructor when we create a logger with a `.` to separate names.
So root of all loggers we created is Logger('ffigen').

Comment (dcharkes) [reply bin/ffigen.dart:14]: 👍

**Pull request merged**